src/protocols/output_management.rs

```diff
@@ -0,0 +1,791 @@
+                    None => {
+                        error!("SetPosition: head is disabled");
+                        return;
+                    }
+                }
+            }
+            zwlr_output_configuration_head_v1::Request::SetTransform { transform } => {
+                let Some(logical) = &mut new_out.logical else {
+                    error!("SetTransform: head is disabled");
+                    return;
+                };
+                logical.transform = match transform {
+                    WEnum::Value(WlTransform::Normal) => Transform::Normal,
+                    WEnum::Value(WlTransform::_90) => Transform::_90,
+                    WEnum::Value(WlTransform::_180) => Transform::_180,
+                    WEnum::Value(WlTransform::_270) => Transform::_270,
+                    WEnum::Value(WlTransform::Flipped) => Transform::Flipped,
+                    WEnum::Value(WlTransform::Flipped90) => Transform::Flipped90,
+                    WEnum::Value(WlTransform::Flipped180) => Transform::Flipped180,
+                    WEnum::Value(WlTransform::Flipped270) => Transform::Flipped270,
+                    _ => {
+                        error!("SetTransform: unknown requested transform");
+                        return;
+                    }
+                }
+            }
+            zwlr_output_configuration_head_v1::Request::SetScale { scale } => {
+                let Some(logical) = &mut new_out.logical else {
+                    error!("SetScale: head is disabled");
+                    return;
+                };
+                logical.scale = scale;
+            }
+            zwlr_output_configuration_head_v1::Request::SetAdaptiveSync { state } => match state {
+                WEnum::Value(AdaptiveSyncState::Enabled) => new_out.vrr_enabled = true,
+                WEnum::Value(AdaptiveSyncState::Disabled) => new_out.vrr_enabled = false,
+                _ => {
+                    error!("SetAdaptativeSync: Unknown requested adaptative sync");
+                    return;
+                }
+            },
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl<D> Dispatch<ZwlrOutputHeadV1, String, D> for OutputManagementManagerState
+where
+    D: GlobalDispatch<ZwlrOutputManagerV1, OutputManagementManagerGlobalData>,
+    D: Dispatch<ZwlrOutputManagerV1, ()>,
+    D: Dispatch<ZwlrOutputHeadV1, String>,
+    D: Dispatch<ZwlrOutputConfigurationV1, OutputConfigurationData>,
+    D: Dispatch<ZwlrOutputConfigurationHeadV1, OutputConfigurationHeadState>,
+    D: Dispatch<ZwlrOutputModeV1, ()>,
+    D: OutputManagementHandler,
+    D: 'static,
+{
+    fn request(
+        state: &mut D,
+        client: &Client,
+        _output_head: &ZwlrOutputHeadV1,
+        request: zwlr_output_head_v1::Request,
+        data: &String,
+        _display: &DisplayHandle,
+        _data_init: &mut DataInit<'_, D>,
+    ) {
+        match request {
+            zwlr_output_head_v1::Request::Release => {
```

Comment:
All handling of release requests should instead be moved to destructors (another fn in Dispatch). This way, even if the client dies, it doesn't linger in the data structures. The release requests themselves can just do nothing.